sample_apps_reinvent2021/go/scheduled-query-sample.go

```diff
@@ -0,0 +1,261 @@
+	utils.HandleError(err, "Failed to subscribe sqs queue to sns topic ", true)
+
+	roleArn, err := timestreamDependencyHelper.CreateIamRole(utils.ROLE_NAME)
+	utils.HandleError(err, fmt.Sprintf("Failed to create iam role with name %s ", utils.ROLE_NAME), true)
+
+	policyArn, _ := timestreamDependencyHelper.CreateIamPolicy(utils.POLICY_NAME)
+	utils.HandleError(err, fmt.Sprintf("Failed to create iam policy with name %s ", utils.POLICY_NAME), true)
+
+	timestreamDependencyHelper.AttachIamPolicy(utils.ROLE_NAME, policyArn)
+	utils.HandleError(err, fmt.Sprintf("Failed to attach iam role with name %s to the policy with name %s ",
+		utils.ROLE_NAME, policyArn), true)
+
+	fmt.Println("Waiting for newly created role to become active")
+	time.Sleep(15 * time.Second)
+
+	//Scheduled Query Activities
+
+	// Create database.
+	err = timestreamBuilder.CreateDatabase(databaseName)
+
+	err = timestreamBuilder.CreateTable(databaseName, tableName, s3BucketName)
+
+	//Create database and table to store scheduled query results
+	err = timestreamBuilder.CreateDatabase(sqResultsDatabaseName)
+
+	err = timestreamBuilder.CreateTable(sqResultsDatabaseName, sqResultsTableName, s3BucketName)
+
+	//err = timestreamBuilder.write_sample_records(*databaseName, *tableName)
+	timestreamBuilder.IngestRecordsFromCsv(*csvFilePath, databaseName, tableName)
+
+	/* Switch between Valid and Invalid Query to test Happy-case and Failure scenarios */
+	if *runInvalidScheduledQueryFlag == false {
+		fmt.Println("Creating a valid Scheduled Query Flow")
+		scheduledQueryArn, err = timestreamBuilder.CreateValidScheduledQuery(topicArn, roleArn, s3BucketName, sqResultsDatabaseName, sqResultsTableName, databaseName, tableName)
+	} else {
+		fmt.Println("Creating a in-valid Scheduled Query Flow")
+		scheduledQueryArn, err = timestreamBuilder.CreateInvalidScheduledQuery(topicArn, roleArn, s3BucketName, sqResultsDatabaseName, sqResultsTableName)
+	}
+	utils.HandleError(err, "Failed to create scheduled query ", true)
+
+	scheduledQueries, err := timestreamBuilder.ListScheduledQueries()
+	if err == nil {
+		fmt.Printf("Total scheduledQueries size : %d\n", len(scheduledQueries))
+	}
+
+	err = timestreamBuilder.DescribeScheduledQuery(scheduledQueryArn)
+	utils.HandleError(err, fmt.Sprintf("Failed to describe scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	err = timestreamBuilder.ExecuteScheduledQuery(scheduledQueryArn, time.Now())
+	utils.HandleError(err, fmt.Sprintf("Failed to execute scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	err = timestreamBuilder.DescribeScheduledQuery(scheduledQueryArn)
+	utils.HandleError(err, fmt.Sprintf("Failed to describe scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	// Sleep for 65 seconds to let ScheduledQuery run
+	fmt.Printf("Waiting for automatic ScheduledQuery executions & notifications from queueUrl %s", queueUrl)
```

Comment:
Include 65 seconds in the waiting message. Else cx will ctrl-c thinking that the app is hung

Reply:
Ack,